### PR TITLE
[#67] Do not generate OSGi "Require-Capability: osgi.ee" header for J…

### PR DESCRIPTION
**Pull request by grgrzybek**
…DK8 compatibility

Signed-off-by: Grzegorz Grzybek <gr.grzybek@gmail.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -173,6 +173,7 @@
                         <DynamicImport-Package>
                             *
                         </DynamicImport-Package>
+                        <_noee>true</_noee>
```
Comment (lukasj) [inline pom.xml:176]: while this is going to work, the right fix is to fix the build (as was done in master) so the maven-bundle-plugin can find and set correct capability in the manifest

Comment (grgrzybek) [reply pom.xml:176]: I don't remember now the exact line of code from bndlib that generates the Require-Capability, but I know it comes from maven-compiler settings. And while JAF needs JDK9+ for module-info.java, I think it'll stay at EE (execution environment) requirement of JDK9...

May I ask what are you referring to in `as was done in master`?

Comment (lukasj) [reply pom.xml:176]: > And while JAF needs JDK9+ for module-info.java, I think it'll stay at EE (execution environment) requirement of JDK9...

Not true. There was some fix on the bnd lib side which - in short - ignores module-info when the value for EE is being generated

> May I ask what are you referring to in as was done in master

to changes which were done after 1.2.2 got released

**Comment by lukasj** [comment]
actually I see no need for this change as correct EE is in MANIFEST since 2.0.0. It makes no sense to remove correct metadata to me

**Comment by grgrzybek** [comment]
@lukasj thanks for clarification. I see correct `Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"` in 2.0.0 which is added by bndlib 4.2.0.
JAF 1.2.2 was created with bndlib 3.5.0.

So my final issue is - is there a chance to release jakarta.activation:jakarta.activation-api 1.2.3 with fixed EE entry? (same for com.sun.activation:jakarta-activation).

**Comment by lukasj** [comment]
No, I do not plan to release 1.2.3. At least not anytime soon.

**Comment by grgrzybek** [comment]
:(
Anything I could do in the name of Apache Karaf community?
There's an [open issue](https://issues.apache.org/jira/browse/KARAF-6703), where Karaf wants to provide consistent set of _features_ to have super-stable, super-consistent set of JavaEE APIs available in an OSGi environment...

**Comment by lukasj** [comment]
well, for 1.2.3 to make sense following needs to be done:
-create EE8 branch from 1.2.2 tag and bump version to 1.2.3-SNAPSHOT
-backport all changes from master, except those renaming javax to jakarta and API changes to this branch
-check TCK

...if you can prepare PR for backports, I should be able to do the rest. 

(btw: I took a look at the open issue you are referencing and one comment - all these javax APIs were removed in JDK 11, not in 9 - some profiles in poms specifies jdk9+ which is simply wrong)

**Comment by grgrzybek** [comment]
Thanks for hints. I'll prepare PR for 1.2.3 after finishing my pending tasks.

Sorry for bothering - which are those "some profiles in poms [that] specifies jdk9+"?
I'm aware of the `java.se.ee` JPMS module which was eventually removed in JDK11, I (if this was my comment) simply treat JDK9 as "non est" and assume that post JDK8 starts with JDK11 (the LTS) :)

**Comment by lukasj** [comment]
the link pointing to cxf-3.4.3/parent/pom.xml#L2274-L2356

**Comment by lukasj** [comment]
for these "old" javax APIs, the most safe option is to completely ignore JDK 9 and 10. On JDK 8, one should use endorsed in most cases, on JDK 11, everything should just work (as with any other 3rd library)

**Comment by grgrzybek** [comment]
I think you're right - I'm doing this anyway (Karaf uses activation API as endorsed library, not as real bundle), so I guess it's ok to just NOT release 1.2.3.
Thanks for your time!

**Pull request closed without merging**